### DISCUSSION TITLE
How to get the final encrypted zip file (with or without compression)

### DISCUSSION THREAD

**Discussion by sntran** [gildas-lormeau/zip.js#452] [Q&A]
Is there a way to get a calculated final size before creating a zip archive for a file with known size? The archive would be encrypted with password, but does not need to be compressed.

**Comment by gildas-lormeau** [comment]
This is not possible if entries are compressed. The total size can be computed for zip files with uncompressed entries though. However, zip.js doesn't offer a function for this, as it isn't necessary.

**Reply by sntran** [reply]
Yes, I mean with uncompressed entries.

Are you open to add such a functionality? Some other tools requires to know the size beforehand, and I would love to be able to stream zip file to them.
